In [1]:
import pandas as pd
import os
import numpy as np
from datetime import timedelta
from datetime import datetime
import timeit

In [2]:
df = pd.read_csv("../../Dados_CET/tercas-quartas-quintas-selecionados/"+ 'tercas' + "/" + "6" +"_selecionado.csv", index_col="Unnamed: 0")

df["Data"] = pd.to_datetime(df["Data"],format='%Y-%m-%d %H:%M:%S', errors='coerce')

df.head()

Lote                Data  Local  Numero Agrupado  Faixa  Entrefaixa  \
0   L3 2018-03-05 23:56:00   5323            10488      2           0   
1   L3 2018-03-05 23:56:00   5323            10488      3           0   
2   L3 2018-03-05 23:56:00   5323            10488      4           0   
3   L3 2018-03-05 23:56:00   5323            10488      4           0   
4   L3 2018-03-05 23:56:00   5323            10488      2           0   

   Registro  Tipo de Registro  Veiculo Final de Placa  Especie  Classe  \
0    202135                 0   351855              4      1.0     0.0   
1    202136                 0   908948              9      1.0     0.0   
2    202137                 0   314820              3      1.0     0.0   
3    202138                 0                              1.0     0.0   
4    202139                 0    97455              6      1.0     0.0   

   Comprimento  Velocidade  Ocupacao  
0         45.0       108.0     766.0  
1         42.0        97.0     835.0  
2         40.0       122.0     656.0  
3         44.0       108.0     772.0  
4         39.0       106.0     761.0

In [3]:
df.Data.dt.date[60].month, df.Data.dt.date[60].day

(3, 6)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 594809 entries, 0 to 594808
Data columns (total 15 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   Lote              594809 non-null  object        
 1   Data              594809 non-null  datetime64[ns]
 2   Local             594809 non-null  int64         
 3   Numero Agrupado   594809 non-null  int64         
 4   Faixa             594809 non-null  int64         
 5   Entrefaixa        594809 non-null  int64         
 6   Registro          594809 non-null  int64         
 7   Tipo de Registro  594809 non-null  int64         
 8   Veiculo           594809 non-null  object        
 9   Final de Placa    594809 non-null  object        
 10  Especie           594218 non-null  float64       
 11  Classe            594218 non-null  float64       
 12  Comprimento       594218 non-null  float64       
 13  Velocidade        594218 non-null  float64       
 14  Ocup

In [5]:
df_base = df.copy()


df_base["Data"] = pd.to_datetime(df_base["Data"],format='%Y-%m-%d %H:%M:%S', errors='coerce')


# limpa a ocupacao
df_base.drop(df_base.loc[df_base['Ocupacao'] =='     ','Ocupacao'].index, inplace=True)
df_base.Ocupacao = df_base.Ocupacao.astype(float)

#dropa vazios
df_base.drop(df_base.loc[df_base['Velocidade'] =='   ','Velocidade'].index, inplace=True)

#dropa não válidos (nan)
df_base.loc[:,'Velocidade'].dropna(inplace=True)

#coloca tudo no mesmo formato de dado int
df_base.loc[:,'Velocidade'] = pd.to_numeric(df_base.loc[:,'Velocidade'])

# colocar faixa no formato int
df_base.loc[:,'Faixa'] = pd.to_numeric(df_base.loc[:,'Faixa'])

# colocar Especie no formato int
df_base.loc[:,'Especie'] = pd.to_numeric(df_base.loc[:,'Especie'])

###### Numero Agrupado é uma String

df_base.reset_index(drop=True,inplace=True)

df_base.head()

Lote                Data  Local  Numero Agrupado  Faixa  Entrefaixa  \
0   L3 2018-03-05 23:56:00   5323            10488      2           0   
1   L3 2018-03-05 23:56:00   5323            10488      3           0   
2   L3 2018-03-05 23:56:00   5323            10488      4           0   
3   L3 2018-03-05 23:56:00   5323            10488      4           0   
4   L3 2018-03-05 23:56:00   5323            10488      2           0   

   Registro  Tipo de Registro  Veiculo Final de Placa  Especie  Classe  \
0    202135                 0   351855              4      1.0     0.0   
1    202136                 0   908948              9      1.0     0.0   
2    202137                 0   314820              3      1.0     0.0   
3    202138                 0                              1.0     0.0   
4    202139                 0    97455              6      1.0     0.0   

   Comprimento  Velocidade  Ocupacao  
0         45.0       108.0     766.0  
1         42.0        97.0     835.0  
2         40.0       122.0     656.0  
3         44.0       108.0     772.0  
4         39.0       106.0     761.0

In [6]:
df_base.Data.dt.date[60].month, df_base.Data.dt.date[60].day

(3, 6)

In [7]:
df_base.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 594809 entries, 0 to 594808
Data columns (total 15 columns):
 #   Column            Non-Null Count   Dtype         
---  ------            --------------   -----         
 0   Lote              594809 non-null  object        
 1   Data              594809 non-null  datetime64[ns]
 2   Local             594809 non-null  int64         
 3   Numero Agrupado   594809 non-null  int64         
 4   Faixa             594809 non-null  int64         
 5   Entrefaixa        594809 non-null  int64         
 6   Registro          594809 non-null  int64         
 7   Tipo de Registro  594809 non-null  int64         
 8   Veiculo           594809 non-null  object        
 9   Final de Placa    594809 non-null  object        
 10  Especie           594218 non-null  float64       
 11  Classe            594218 non-null  float64       
 12  Comprimento       594218 non-null  float64       
 13  Velocidade        594218 non-null  float64       
 14  Ocup

In [8]:
df_agregado = df_base.copy()

df_agregado.loc[:,'V1'] = df_agregado.loc[:,'Velocidade'].copy()
df_agregado.loc[:,'V2'] = df_agregado.loc[:,'Velocidade'].copy()
df_agregado.loc[:,'Oc1'] = df_agregado.loc[:,'Ocupacao'].copy()
df_agregado.loc[:,'Oc2'] = df_agregado.loc[:,'Ocupacao'].copy()



df_agregado = df_agregado.groupby(['Numero Agrupado',"Faixa",'Especie',pd.Grouper(key='Data', freq='4Min')]).agg(
                                                                        {"Registro": "count",
                                                                        'Velocidade': 'mean',
                                                                        'V1':'median',
                                                                        'V2':'std',
                                                                        'Ocupacao':'mean',
                                                                        'Oc1':'median',
                                                                        'Oc2':'std',
                                                                        })
df_agregado.columns=[
            "Volume",
            'Vel_media',
            'Vel_mediana',
            'Vel_desvpad',
            'Ocu_media',
            'Ocu_mediana',
            'Ocu_desvpad'
            ] # Renomeia

df_agregado.reset_index(inplace=True)
df_agregado.head()

Numero Agrupado  Faixa  Especie                Data  Volume  Vel_media  \
0            10433      1      0.0 2018-03-05 23:56:00       1      144.0   
1            10433      1      0.0 2018-03-06 00:00:00       1      122.0   
2            10433      1      0.0 2018-03-06 00:12:00       1      103.0   
3            10433      1      0.0 2018-03-06 00:32:00       1      117.0   
4            10433      1      0.0 2018-03-06 00:52:00       1      117.0   

   Vel_mediana  Vel_desvpad  Ocu_media  Ocu_mediana  Ocu_desvpad  
0        144.0          NaN      346.0        346.0          NaN  
1        122.0          NaN      392.0        392.0          NaN  
2        103.0          NaN      346.0        346.0          NaN  
3        117.0          NaN      257.0        257.0          NaN  
4        117.0          NaN      385.0        385.0          NaN

In [9]:
df_agregado.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23291 entries, 0 to 23290
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   Numero Agrupado  23291 non-null  int64         
 1   Faixa            23291 non-null  int64         
 2   Especie          23291 non-null  float64       
 3   Data             23291 non-null  datetime64[ns]
 4   Volume           23291 non-null  int64         
 5   Vel_media        23291 non-null  float64       
 6   Vel_mediana      23291 non-null  float64       
 7   Vel_desvpad      17700 non-null  float64       
 8   Ocu_media        23291 non-null  float64       
 9   Ocu_mediana      23291 non-null  float64       
 10  Ocu_desvpad      17700 non-null  float64       
dtypes: datetime64[ns](1), float64(7), int64(3)
memory usage: 2.0 MB


In [10]:
set(df_agregado["Numero Agrupado"])

{10433, 10482, 10488, 10500, 10531, 10557, 10595, 10761}

In [11]:
df_agregado.head(50)

Numero Agrupado  Faixa  Especie                Data  Volume   Vel_media  \
0             10433      1      0.0 2018-03-05 23:56:00       1  144.000000   
1             10433      1      0.0 2018-03-06 00:00:00       1  122.000000   
2             10433      1      0.0 2018-03-06 00:12:00       1  103.000000   
3             10433      1      0.0 2018-03-06 00:32:00       1  117.000000   
4             10433      1      0.0 2018-03-06 00:52:00       1  117.000000   
5             10433      1      0.0 2018-03-06 01:08:00       1  139.000000   
6             10433      1      0.0 2018-03-06 01:28:00       1  136.000000   
7             10433      1      0.0 2018-03-06 02:04:00       1  117.000000   
8             10433      1      0.0 2018-03-06 03:08:00       1  131.000000   
9             10433      1      0.0 2018-03-06 03:44:00       1  106.000000   
10            10433      1      0.0 2018-03-06 04:16:00       1  119.000000   
11            10433      1      0.0 2018-03-06 04:32:00       1  114.000000   
12            10433      1      0.0 2018-03-06 04:56:00       1  119.000000   
13            10433      1      0.0 2018-03-06 05:04:00       1  119.000000   
14            10433      1      0.0 2018-03-06 05:08:00       2  112.500000   
15            10433      1      0.0 2018-03-06 05:16:00       1  125.000000   
16            10433      1      0.0 2018-03-06 05:24:00       1  114.000000   
17            10433      1      0.0 2018-03-06 05:28:00       1  131.000000   
18            10433      1      0.0 2018-03-06 05:32:00       1  139.000000   
19            10433      1      0.0 2018-03-06 05:40:00       3  120.333333   
20            10433      1      0.0 2018-03-06 05:48:00       2  122.000000   
21            10433      1      0.0 2018-03-06 05:52:00       2  120.500000   
22            10433      1      0.0 2018-03-06 05:56:00       2  146.000000   
23            10433      1      0.0 2018-03-06 06:00:00       1  144.000000   
24            10433      1      0.0 2018-03-06 06:04:00       3  118.666667   
25            10433      1      0.0 2018-03-06 06:24:00       1   97.000000   
26            10433      1      0.0 2018-03-06 06:28:00       3  124.666667   
27            10433      1      0.0 2018-03-06 06:36:00       2  106.500000   
28            10433      1      0.0 2018-03-06 06:40:00       4  131.750000   
29            10433      1      0.0 2018-03-06 06:44:00       3  120.333333   
30            10433      1      0.0 2018-03-06 06:48:00       2  135.000000   
31            10433      1      0.0 2018-03-06 06:52:00       8  123.500000   
32            10433      1      0.0 2018-03-06 06:56:00       8  122.250000   
33            10433      1      0.0 2018-03-06 07:00:00      16  126.187500   
34            10433      1      0.0 2018-03-06 07:04:00      10  120.000000   
35            10433      1      0.0 2018-03-06 07:08:00      10  118.600000   
36            10433      1      0.0 2018-03-06 07:12:00      20  125.100000   
37            10433      1      0.0 2018-03-06 07:16:00      24  121.625000   
38            10433      1      0.0 2018-03-06 07:20:00      12  120.916667   
39            10433      1      0.0 2018-03-06 07:24:00      12  128.666667   
40            10433      1      0.0 2018-03-06 07:28:00      14  117.500000   
41            10433      1      0.0 2018-03-06 07:32:00      21  128.857143   
42            10433      1      0.0 2018-03-06 07:36:00      31  123.548387   
43            10433      1      0.0 2018-03-06 07:40:00      17  121.705882   
44            10433      1      0.0 2018-03-06 07:44:00      24  125.625000   
45            10433      1      0.0 2018-03-06 07:48:00      17  126.235294   
46            10433      1      0.0 2018-03-06 07:52:00      31  122.064516   
47            10433      1      0.0 2018-03-06 07:56:00      20  119.800000   
48            10433      1      0.0 2018-03-06 08:00:00      17  127.294118   
49            10433      1      0.0 2018-03-06 08:04:0

In [12]:
df_agregado.Data.dt.date[6] + timedelta(days=1)

datetime.date(2018, 3, 7)

In [13]:
list(set(df_agregado["Numero Agrupado"])), len(list(set(df_agregado["Numero Agrupado"])))

([10433, 10531, 10500, 10595, 10761, 10482, 10488, 10557], 8)

In [16]:
list(set(df_agregado["Especie"])), len(list(set(df_agregado["Especie"])))

([0.0, 1.0, 2.0, 3.0], 4)

In [22]:
list(set(df_agregado["Faixa"])), len(list(set(df_agregado["Faixa"])))

([1, 2, 3, 4, 5], 5)

In [23]:
print("como tem um dataframe para cada combinacao de radar/especia/faixa sao, no maximo, {} subsets no dataframe".format(8*5*4))

print("'no maximo' por que combinacoes que nao tem dados, por exemplo: um radar de uma rua nao a 5a faixa ")

como tem um dataframe para cada combinacao de radar/especia/faixa sao, no maximo, 160 subsets no dataframe
'no maximo' por que combinacoes que nao tem dados, por exemplo: um radar de uma rua nao a 5a faixa 


### como tem um dataframe para cada combinacao de radar/especia/faixa sao, no maximo, 160 subsets no dataframe
### 'no maximo' por que combinacoes que nao tem dados, por exemplo: um radar de uma rua nao a 5a faixa 

In [24]:
(24*60/4)

360.0

In [25]:
%%time

df_final = pd.DataFrame()


df_reduzido = df_agregado.copy()


for radar in list(set(df_agregado["Numero Agrupado"])):#[ 10761]


    for especie in list(set(df_reduzido["Especie"])):#[1.0]


        for faixa in list(set(df_reduzido["Faixa"])):#[1]
            # print(radar, especie, faixa)
            
            # faz um pequeno subset do df buscando criar uma configuração de radar, faixa e especie especifica
            radar_especifico = df_reduzido.loc[(df_reduzido["Numero Agrupado"] == radar)
                                            &   (df_reduzido["Especie"] == especie)
                                            &   (df_reduzido["Faixa"] == faixa)].sort_values(by="Data").reset_index(drop=True)




            df_aux = pd.DataFrame()

            ## Esse try existe para caso uma determinada combinacao de radar/especie/faixa 
            ## nao exista no df
            try:
                # Coleta as datas do dataframe para criar o novo datetime index
                dia_inicio = radar_especifico.Data.dt.date[1]
                dia_fim = radar_especifico.Data.dt.date[1] + timedelta(days=1)

                # cria o datetime index, com frequencia desejada
                index_data = pd.date_range(dia_inicio, dia_fim, freq="4 min", closed="left") 
                df_aux["Data"] = index_data

                # Faz o merge dos dataframes auxiliar com o do radar especifico, respeitando o datetime index (left)
                df_aux_1 = df_aux.merge(radar_especifico, on='Data', how="left")



                # o seguinte preenchimento deve ser feito apos as novas linhas de 4 minutos que
                # que nao tem registros serem adicionadas, e 
                # somente nas colunas selecionadas, que sao as que sabemos os parametros

                colunas_preenchiveis = ["Numero Agrupado","Faixa","Especie"]
                df_aux_1[colunas_preenchiveis] = df_aux_1[colunas_preenchiveis].fillna(method="ffill")
                df_aux_1[colunas_preenchiveis] = df_aux_1[colunas_preenchiveis].fillna(method="bfill")

                df_final = pd.concat([df_final, df_aux_1])
                df_final.reset_index(drop=True, inplace=True)

            except:
                pass


df_final

Wall time: 2.21 s


Data  Numero Agrupado  Faixa  Especie  Volume  Vel_media  \
0     2018-03-06 00:00:00          10433.0    1.0      0.0     1.0      122.0   
1     2018-03-06 00:04:00          10433.0    1.0      0.0     NaN        NaN   
2     2018-03-06 00:08:00          10433.0    1.0      0.0     NaN        NaN   
3     2018-03-06 00:12:00          10433.0    1.0      0.0     1.0      103.0   
4     2018-03-06 00:16:00          10433.0    1.0      0.0     NaN        NaN   
...                   ...              ...    ...      ...     ...        ...   
44635 2018-03-06 23:40:00          10557.0    4.0      3.0     NaN        NaN   
44636 2018-03-06 23:44:00          10557.0    4.0      3.0     NaN        NaN   
44637 2018-03-06 23:48:00          10557.0    4.0      3.0     NaN        NaN   
44638 2018-03-06 23:52:00          10557.0    4.0      3.0     NaN        NaN   
44639 2018-03-06 23:56:00          10557.0    4.0      3.0     NaN        NaN   

       Vel_mediana  Vel_desvpad  Ocu_media  Ocu_mediana  Ocu_desvpad  
0            122.0          NaN      392.0        392.0          NaN  
1              NaN          NaN        NaN          NaN          NaN  
2              NaN          NaN        NaN          NaN          NaN  
3            103.0          NaN      346.0        346.0          NaN  
4              NaN          NaN        NaN          NaN          NaN  
...            ...          ...        ...          ...          ...  
44635          NaN          NaN        NaN          NaN          NaN  
44636          NaN          NaN        NaN          NaN          NaN  
44637          NaN          NaN        NaN          NaN          NaN  
44638          NaN          NaN        NaN          NaN          NaN  
44639          NaN          NaN        NaN          NaN          NaN  

[44640 rows x 11 columns]

In [26]:
df_final.iloc[359:721]

Data  Numero Agrupado  Faixa  Especie  Volume  Vel_media  \
359 2018-03-06 23:56:00          10433.0    1.0      0.0     NaN        NaN   
360 2018-03-06 00:00:00          10433.0    2.0      0.0     NaN        NaN   
361 2018-03-06 00:04:00          10433.0    2.0      0.0     NaN        NaN   
362 2018-03-06 00:08:00          10433.0    2.0      0.0     NaN        NaN   
363 2018-03-06 00:12:00          10433.0    2.0      0.0     NaN        NaN   
..                  ...              ...    ...      ...     ...        ...   
716 2018-03-06 23:44:00          10433.0    2.0      0.0     NaN        NaN   
717 2018-03-06 23:48:00          10433.0    2.0      0.0     NaN        NaN   
718 2018-03-06 23:52:00          10433.0    2.0      0.0     NaN        NaN   
719 2018-03-06 23:56:00          10433.0    2.0      0.0     NaN        NaN   
720 2018-03-06 00:00:00          10433.0    3.0      0.0     NaN        NaN   

     Vel_mediana  Vel_desvpad  Ocu_media  Ocu_mediana  Ocu_desvpad  
359          NaN          NaN        NaN          NaN          NaN  
360          NaN          NaN        NaN          NaN          NaN  
361          NaN          NaN        NaN          NaN          NaN  
362          NaN          NaN        NaN          NaN          NaN  
363          NaN          NaN        NaN          NaN          NaN  
..           ...          ...        ...          ...          ...  
716          NaN          NaN        NaN          NaN          NaN  
717          NaN          NaN        NaN          NaN          NaN  
718          NaN          NaN        NaN          NaN          NaN  
719          NaN          NaN        NaN          NaN          NaN  
720          NaN          NaN        NaN          NaN          NaN  

[362 rows x 11 columns]

In [27]:
df_final.shape[0]/360

124.0

### Portanto, dos 160 subsets possiveis, somente 124 estao presentes

In [28]:
tercas = ["6","13","20","27"]
quartas = ["7","14","21","28"]
quintas = ["1","8","15","22","29"]

In [30]:
%%time


for i,DOW in enumerate([    ("tercas", tercas),
                            ("quartas", quartas),
                            ("quintas", quintas)     ]):
    # print(i, DOW)

    for day in DOW[1]:
        # print(day) ## day is a string
        start = timeit.timeit()

        # new_df = pd.read_csv("../../Dados_CET/Marco_2018_nAg/"+ day +"_nAg.csv", index_col="Unnamed: 0")
        print("reading into     " + "../../Dados_CET/tercas-quartas-quintas-selecionados/"+ DOW[0] + "/" + day +"_selecionado.csv")

        # print("../../Dados_CET/Marco_2018_nAg/"+ day +"_nAg.csv")
        df = pd.read_csv("../../Dados_CET/tercas-quartas-quintas-selecionados/"+ DOW[0] + "/" + day +"_selecionado.csv")


        # print("../../Dados_CET/tercas-quartas-quintas-selecionados/"+ DOW[0] + "/" + day +"_selecionado.csv")
        # df_selected = df[df["Numero Agrupado"].isin(radares)].reset_index(drop=True)

        df["Data"] = pd.to_datetime(df["Data"],format='%Y-%m-%d %H:%M:%S', errors='coerce')


        df_base = df.copy()


        # limpa a ocupacao
        df_base.drop(df_base.loc[df_base['Ocupacao'] =='     ','Ocupacao'].index, inplace=True)
        df_base.Ocupacao = df_base.Ocupacao.astype(float)

        #dropa vazios
        df_base.drop(df_base.loc[df_base['Velocidade'] =='   ','Velocidade'].index, inplace=True)

        #dropa não válidos (nan)
        df_base.loc[:,'Velocidade'].dropna(inplace=True)

        #coloca tudo no mesmo formato de dado int
        df_base.loc[:,'Velocidade'] = pd.to_numeric(df_base.loc[:,'Velocidade'])

        # colocar faixa no formato int
        df_base.loc[:,'Faixa'] = pd.to_numeric(df_base.loc[:,'Faixa'])

        # colocar Especie no formato int
        df_base.loc[:,'Especie'] = pd.to_numeric(df_base.loc[:,'Especie'])

        ###### Numero Agrupado é uma int

        df_base.reset_index(drop=True,inplace=True)


        df_agregado = df_base.copy()

        df_agregado.loc[:,'V1'] = df_agregado.loc[:,'Velocidade'].copy()
        df_agregado.loc[:,'V2'] = df_agregado.loc[:,'Velocidade'].copy()
        df_agregado.loc[:,'Oc1'] = df_agregado.loc[:,'Ocupacao'].copy()
        df_agregado.loc[:,'Oc2'] = df_agregado.loc[:,'Ocupacao'].copy()



        df_agregado = df_agregado.groupby(['Numero Agrupado',"Faixa",'Especie',pd.Grouper(key='Data', freq='4Min')]).agg(
                                                                                {"Registro": "count",
                                                                                'Velocidade': 'mean',
                                                                                'V1':'median',
                                                                                'V2':'std',
                                                                                'Ocupacao':'mean',
                                                                                'Oc1':'median',
                                                                                'Oc2':'std',
                                                                                })
        df_agregado.columns=[
                    "Volume",
                    'Vel_media',
                    'Vel_mediana',
                    'Vel_desvpad',
                    'Ocu_media',
                    'Ocu_mediana',
                    'Ocu_desvpad'
                    ] # Renomeia

        df_agregado.reset_index(inplace=True)




        df_final = pd.DataFrame()


        df_reduzido = df_agregado.copy()


        for radar in list(set(df_reduzido["Numero Agrupado"])):#[ 10761]


            for especie in list(set(df_reduzido["Especie"])):#[1.0]


                for faixa in list(set(df_reduzido["Faixa"])):#[1]
                    # print(radar, especie, faixa)
                    
                    # faz um pequeno subset do df buscando criar uma configuração de radar, faixa e especie especifica
                    radar_especifico = df_reduzido.loc[(df_reduzido["Numero Agrupado"] == radar)
                                                    &   (df_reduzido["Especie"] == especie)
                                                    &   (df_reduzido["Faixa"] == faixa)].sort_values(by="Data").reset_index(drop=True)




                    df_aux = pd.DataFrame()

                    ## Esse try existe para caso uma determinada combinacao de radar/especie/faixa 
                    ## nao exista no df
                    try:
                        # Coleta as datas do dataframe para criar o novo datetime index
                        dia_inicio = radar_especifico.Data.dt.date[1]
                        dia_fim = radar_especifico.Data.dt.date[1] + timedelta(days=1)

                        # cria o datetime index, com frequencia desejada
                        index_data = pd.date_range(dia_inicio, dia_fim, freq="4 min", closed="left") 
                        df_aux["Data"] = index_data

                        # Faz o merge dos dataframes auxiliar com o do radar especifico, respeitando o datetime index (left)
                        df_aux_1 = df_aux.merge(radar_especifico, on='Data', how="left")



                        # o seguinte preenchimento deve ser feito apos as novas linhas de 4 minutos que
                        # que nao tem registros serem adicionadas, e 
                        # somente nas colunas selecionadas, que sao as que sabemos os parametros

                        colunas_preenchiveis = ["Numero Agrupado","Faixa","Especie"]
                        df_aux_1[colunas_preenchiveis] = df_aux_1[colunas_preenchiveis].fillna(method="ffill")
                        df_aux_1[colunas_preenchiveis] = df_aux_1[colunas_preenchiveis].fillna(method="bfill")

                        df_final = pd.concat([df_final, df_aux_1])
                        df_final.reset_index(drop=True, inplace=True)

                    except:
                        pass






        print()
        print("Saving to    " + "../../Dados_CET/tercas-quartas-quintas-selecionados/"+ DOW[0] + "-agrupado/" + day +"_agrupado.csv")

        df_final.to_csv("../../Dados_CET/tercas-quartas-quintas-selecionados/"+ DOW[0] + "-agrupado/" + day +"_agrupado.csv")

        end = timeit.timeit()
        print(end - start)

        print()
        print()
        print()


reading into     ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas/6_selecionado.csv

Saving to    ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas-agrupado/6_agrupado.csv
-0.0060570000005100155



reading into     ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas/13_selecionado.csv

Saving to    ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas-agrupado/13_agrupado.csv
-0.003229500000088592



reading into     ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas/20_selecionado.csv

Saving to    ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas-agrupado/20_agrupado.csv
0.017152599999462836



reading into     ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas/27_selecionado.csv

Saving to    ../../Dados_CET/tercas-quartas-quintas-selecionados/tercas-agrupado/27_agrupado.csv
-0.004349599998022313



reading into     ../../Dados_CET/tercas-quartas-quintas-selecionados/quartas/7_selecionado.csv

Saving to    ../../Dados_CET/terca